src/azure-cli/azure/cli/command_modules/cdn/commands.py

```diff
@@ -136,3 +151,39 @@ def _inner_not_found(ex):
 
     with self.command_group('cdn edge-node', cdn_edge_sdk) as g:
         g.command('list', 'list')
+
+    with self.command_group('cdn waf policy', cdn_waf_policy_sdk, is_preview=True) as g:
+        g.show_command('show', 'get')
+        g.command('list', 'list')
+        g.custom_command('set', 'set_waf_policy', client_factory=cf_waf_policy)
+        g.command('delete', 'delete')
+
+    with self.command_group('cdn waf policy managed-rule-set', cdn_waf_policy_sdk, is_preview=True) as g:
+        g.custom_command('add', 'add_waf_policy_managed_rule_set', client_factory=cf_waf_policy)
+        g.custom_command('remove', 'remove_waf_policy_managed_rule_set', client_factory=cf_waf_policy)
+        g.custom_command('list', 'list_waf_policy_managed_rule_sets', client_factory=cf_waf_policy)
+        g.custom_show_command('show', 'show_waf_policy_managed_rule_set', client_factory=cf_waf_policy)
+        g.custom_command('list-available', 'list_waf_managed_rule_set', client_factory=cf_waf_rule_set)
+        g.custom_command('show-available', 'show_waf_managed_rule_set', client_factory=cf_waf_rule_set)
```

Comment:
`list-available` can be used for the scenario to list all available resources. What do you mean for `show-available`? Typically we only use `show` command to show the specified resource. May I know why there is a `show-available` command?

Reply:
The show available command shows one of the available managed rule sets, retrieved by name. Managed rule sets are deeply nested resources, and it is helpful to retrieve one individually for inspecting available rule groups and rules within the managed rule set rather than sifting through the list of all of them. Please let me know if there is some issue with `show-available` and I can just remove it for now in order to expedite this release.

Reply:
I think you want to show all available rule groups in a specific rule set, right? 
In this way, you can make your command design like `az cdn waf policy managed-rule-set group list` to show all rule groups information in a rule set.

Reply:
Unfortunately, our service is a little confusing because available managed rule sets, available managed rule groups, and their available managed rules are separate from the managed rule sets with their rule group overrides and rule overrides that are applied to a particular policy. For this reason it's difficult to differentiate between whether an action is related to an available managed rule set/group vs an applied managed rule set/group. I've removed the `show-available` commands for the `cdn waf policy managed-rule-set` and `cdn waf policy managed-rule-set rule-group` command groups since it isn't essential for our service and is causing confusion.